pkg/network/msg.go

```diff
@@ -75,3 +91,26 @@ type DataSent struct {
 	ID   string          `json:"id"`
 	Data []entities.Data `json:"data"`
 }
+
+// NewMessage creates a message
+func NewMessage(msg interface{}) MessageInterface {
+	return Message{Payload: msg}
+}
+
+// Serialize serializes the message in a byte stream
+func (m Message) Serialize() ([]byte, error) {
+	data, err := json.Marshal(&m.Payload)
+	if err != nil {
+		return nil, fmt.Errorf("error econding JSON message: %w", err)
+	}
+	return data, nil
```

Comment:
I agree that this is cleaner and reduces code (which is awesome... more code, more vulnerable we are in lots of aspects), but shouldn't the error be thrown on the layer it happened? For example, if `msg.go` is expected to return a marshaled payload and it fails to do so, shouldn't it throw the error, instead of it's caller?

Reply:
In that case I think we could still wrapping the error to add some context to it. What dou you think? @ramonhpr @netoax @lcbm @Igoraamc

Reply:
@esdrasjnr I think we can continue wrapping this error.

Reply:
These changes can be postponed to a future PR, so I will integrate this PR.
time :p